app/src/main/java/org/kiwix/kiwixmobile/webserver/KiwixServer.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2020 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.webserver
+
+import android.util.Log
+import org.kiwix.kiwixlib.JNIKiwixException
+import org.kiwix.kiwixlib.JNIKiwixServer
+import org.kiwix.kiwixmobile.nav.destination.library.LibraryFactory
+import javax.inject.Inject
+
+private const val TAG = "KiwixServer"
+
+class KiwixServer @Inject constructor(kiwixLibraryFactory: LibraryFactory) {
+  private val libFactory = kiwixLibraryFactory
```

Comment:
this will disappear but I should highlight this is a mistake, if you just moved `private val` to the constructor parameter then you wouldn't need to assign it to this variable